src/transaction/declare_v2.rs

```diff
@@ -261,15 +323,15 @@ impl DeclareV2 {
         &self,
         state: &mut S,
     ) -> Result<(), TransactionError> {
-        let casm_class = self
-            .casm_class
-            .get_or_try_init(|| {
-                CasmContractClass::from_contract_class(self.sierra_contract_class.clone(), true)
-            })
-            .map_err(|e| TransactionError::SierraCompileError(e.to_string()))?;
-
-        state.set_compiled_class_hash(&self.sierra_class_hash, &self.compiled_class_hash)?;
-        state.set_compiled_class(&self.compiled_class_hash, casm_class.clone())?;
+        let casm_class = if self.casm_class.is_some() {
+            self.casm_class.as_ref().unwrap().clone()
+        } else {
+            CasmContractClass::from_contract_class(self.sierra_contract_class.clone(), true)
+                .map_err(|e| TransactionError::SierraCompileError(e.to_string()))?
+        };
```

Comment:
I think this would look cleaner with a `match`.

Reply:
this ^^

Reply:
fixed